src/client/datascience/baseJupyterSession.ts

```diff
@@ -171,6 +171,7 @@ export abstract class BaseJupyterSession implements IJupyterSession {
     public async restart(): Promise<void> {
         if (this.session?.isRemoteSession && this.session.kernel) {
             await this.session.kernel.restart();
+            this.setSession(this.session, true);
```

Comment:
A new websocket is created, we need to ensure this new websocket connection is used (new argument)